presto-orc/src/main/java/com/facebook/presto/orc/reader/NullWrappingColumnReader.java

```diff
@@ -59,17 +60,18 @@ protected void makeInnerQualifyingSet()
             return;
         }
         hasNulls = true;
+        boolean nonDeterministic = filter != null && !deterministicFilter;
         if (innerQualifyingSet == null) {
             innerQualifyingSet = new QualifyingSet();
         }
         int[] inputRows = inputQualifyingSet.getPositions();
         int numActive = inputQualifyingSet.getPositionCount();
 
         innerQualifyingSet.reset(numActive);
-        int prevRow = posInRowGroup;
+        int prevRow = 0;
         int prevInner = innerPosInRowGroup;
         numNullsToAdd = 0;
-        boolean keepNulls = filter == null || filter.testNull();
+        boolean keepNulls = filter == null || (!nonDeterministic && filter.testNull());
```

Comment:
nonDeterministic variable is used only once; inlining simplifies the code to:

```
        boolean keepNulls = filter == null || (deterministicFilter && filter.testNull());
```

However, there must be logic to handle nulls for non-deterministic filters and I don't see where it is. Could you give me a pointer?